jsonschema-module-jackson/src/main/java/com/github/victools/jsonschema/module/jackson/AnnotationHelper.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2024 VicTools.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.victools.jsonschema.module.jackson;
+
+import com.fasterxml.classmate.members.ResolvedMember;
+import com.fasterxml.jackson.annotation.JacksonAnnotationsInside;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+final class AnnotationHelper {
+
+    static final Predicate<Annotation> JACKSON_ANNOTATIONS_INSIDE_ANNOTATED_FILTER = (annotation) ->
+            annotation.annotationType().isAnnotationPresent(JacksonAnnotationsInside.class);
+
+    private AnnotationHelper() {
+        super();
+    }
+
+    /**
+     * Resolves the specified annotation on the given resolved member and resolve indirect jackson annotations.
+     *
+     * <p>It uses the same algorithm as
+     * {@link com.github.victools.jsonschema.generator.TypeContext#getAnnotationFromList(Class, List, Predicate)}
```

Comment:
**question**: If it is the same logic as in the `TypeContext`, why redundantly implement again instead of using that existing code?
I mean, that's the whole point of the method being in the `TypeContext` class. To make it re-usable and at the same time allow extending it if need be, since it's _not_ inside a `static` class. This feels like we're locking it down too much (in this module) on top of it being duplication and not allowing other modules to use it at the same time (inviting even more code duplication down the line).

Perhaps move the `AnnotationHelper` class into the main generator, extract that existing logic from the `TypeContext`, and include it here to (leaving the method in the `TypeContext` merely as delegate to this helper class' equivalent method). What do you reckon?

The only jackson-specific thing left is then the `Predicate` at the top.

Reply:
To be totally fair that was my initial thought but I figured that the first step was to have a working principle and to make sure I wasn't breaking any Jackson bound logic.
If you're fine with it I would be glad to do it. But maybe tomorrow (late here too 🤭)

Reply:
Done in https://github.com/victools/jsonschema-generator/pull/487/commits/d559d8c11def38a14f39cd34a1cba3dfcf9516a2 and https://github.com/victools/jsonschema-generator/pull/487/commits/e6e53b831031cce73912f9b4c6e628e3191ad4d2